coderd/httpapi/nilmap.go

```diff
@@ -0,0 +1,72 @@
+package httpapi
+
+import (
+	"reflect"
+
+	"golang.org/x/xerrors"
+)
+
+func ContainsNilMap(v any) error {
+	visited := make(map[uintptr]bool)
+	if hasNil, field := findNilMapsRec(reflect.ValueOf(v), visited); hasNil {
+		ty := reflect.TypeOf(v)
+		extra := ""
+		if field != "" {
+			extra = " in field " + field
+		}
+		return xerrors.Errorf("nil maps received in type %q%s", ty.String(), extra)
+	}
+	return nil
+}
```

Comment:
I'd like us to also check that this is never called outside of a testing context and return early.

Reply:
Good extra guard